express-post-json/server.ts

```diff
@@ -0,0 +1,44 @@
+import express from 'express';
+
+type Grade = {
+  id: number;
+  name: string;
+  course: string;
+  score: number;
+};
+
+type NextId = number;
+
+const grades: Record<number, Grade> = {};
+
+let nextId: NextId = 1;
+
+const app = express();
+
+app.get('/api/grades', (req, res) => {
+  const localArr: Grade[] = Object.values(grades);
+  res.json(localArr);
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  const grade = req.body;
+  grade.id = nextId;
+  grades[nextId] = grade;
+  nextId++;
+
+  if (
+    typeof !grade.name === 'string' ||
+    typeof !grade.course === 'string' ||
+    typeof !grade.score === 'number'
+  ) {
+    res.status(204);
```

Comment:
This is a client error so you should send 400. This doesn't actually send a response, though. You need to use `sendStatus(400)` (or `res.status(400).send('message')`) to have it actually send.

You also need to return after this so that the other calls at line 38 and 39 are not executed. Otherwise, you'll send a response twice.